### PR TITLE
add tools-ansible playbook

### PR DESCRIPTION
**Pull request by holys**
After deployed, it might looks like:

```
.
├── backup
├── bin
│   ├── loader
│   ├── mydumper
│   └── syncer
├── conf
│   ├── loader_1.toml
│   └── syncer_1.toml
├── data
│   └── mydumper_1
│       ├── foo.bar-schema.sql
│       ├── foo.bar.sql
│       ├── foo-schema-create.sql
│       └── metadata
├── log
│   ├── loader_1.log
│   ├── loader_1_stderr.log
│   ├── syncer_1.log
│   └── syncer_1_stderr.log
├── meta
│   └── syncer_1.meta
└── scripts
    ├── run_integrated_1.sh
    ├── run_loader_1.sh
    ├── run_syncer_1.sh
    ├── start_integrated_1.sh
    ├── start_loader_1.sh
    ├── start_syncer_1.sh
    ├── stop_integrated_1.sh
    ├── stop_loader_1.sh
    └── stop_syncer_1.sh

```

### CODE REVIEW

File: tools-ansible/roles/download/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+- name: set outbound network fact
+  set_fact: has_outbound_network={{ outbound_network_st.stdout.strip() == '0' }}
+
+- fail:
+    msg: "The Control Machine must have access to the Internet in order to download tidb-tools packages"
+  when: not has_outbound_network
+
+- name: detect GFW
+  shell: >
```
Comment (LinuxGit) [inline tools-ansible/roles/download/tasks/main.yml]: remove gfw check

File: tools-ansible/ansible.cfg [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+# cow_selection = random
+
+stdout_callback = skippy
+
+# log information about executions at the designated location
+log_path = log/ansible.log
+
+deprecation_warnings = False
+
+callback_whitelist = help
```
Comment (LinuxGit) [inline tools-ansible/ansible.cfg]: rely on https://github.com/pingcap/tidb-ansible/blob/master/callback_plugins/help.py

File: tools-ansible/group_vars/all.yml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+log_dir: "{{ deploy_dir }}/log"
+
+# Local
+ansible_log_dir: "{{ playbook_dir }}/log"
+downloads_dir: "{{ playbook_dir }}/downloads"
+conf_dir: "{{ playbook_dir }}/conf"
+resources_dir: "{{ playbook_dir }}/resources"
+script_dir:    "{{ playbook_dir }}/scripts"
+binary_dir:    "{{ playbook_dir }}/resources/bin"
+fetch_tmp_dir: "{{ playbook_dir }}/fetch_tmp"
```
Comment (LinuxGit) [inline tools-ansible/group_vars/all.yml]: remove fetch_tmp_dir and fetch_log_dir

File: tools-ansible/hosts.ini [hunk 4]
```diff
@@ -0,0 +1,8 @@
+[servers]
```
Comment (LinuxGit) [inline tools-ansible/hosts.ini]: remove hosts.ini file

File: tools-ansible/roles/integrated/tasks/main.yml [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    backup: yes
+  with_items:
+    - run
+  vars:
+    role_status_dir: status/{{ role_name }}/{{ job_name }}
+  register: integrated_script
+
+- name: backup script file
+  command: mv "{{ integrated_script.backup_file }}" "{{ backup_dir }}"
+  when: integrated_script.changed and integrated_script.backup_file is defined
```
Comment (LinuxGit) [inline tools-ansible/roles/integrated/tasks/main.yml]: when:
    - integrated_script.changed
    - integrated_script.backup_file is define

File: tools-ansible/roles/loader/templates/run_loader_binary.sh.j2 [hunk 6]
```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
+set -e
+
+ulimit -n 1000000
```
Comment (LinuxGit) [inline tools-ansible/roles/loader/templates/run_loader_binary.sh.j2:4]: make sure the system have set ulimit

File: tools-ansible/inventory.ini [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+
+## Global variables
+
+[all:vars]
+# we will use deploy_dir/data to save mydumper data, please ensure deploy_dir is large enough to hold the dump data.
+deploy_dir = /home/tidb/deploy
+# ssh via normal user
+ansible_user = tidb
+
+# timezone of deployment region
```
Comment (LinuxGit) [inline tools-ansible/inventory.ini]: remove useless variable:
+# timezone of deployment region
 +timezone = Asia/Shanghai
 +set_timezone = True
 +
 +enable_ntpd = True
 +set_hostname = False

File: tools-ansible/roles/loader/templates/run_loader_binary.sh.j2 (hunk 6 above)
Comment (holys) [reply tools-ansible/roles/loader/templates/run_loader_binary.sh.j2:4]: done

**Comment by iamxy** [comment]
Is it necessary to make Loader as a systemd service?

**Comment by holys** [comment]
It may take a very long time to load data for loader, so I think is reasonable to make loader as a systemd service. Besides, that can avoid hangup exits.

**Pull request closed without merging**